browser/js/app.js

```diff
@@ -73,19 +66,34 @@ var player = {
 
 var otherPlayers = [];
 
+socket.on('otherPlayerJoin', function (otherPlayerData) {
+    console.log(otherPlayerData.id + ' has joined!');
+    otherPlayerData.sprite = new Sprite('img/capguy-walk.png', [0, 0], [184, 325], 16, [0, 1, 2, 3, 4, 5, 6, 7]);
+    otherPlayers.push(otherPlayerData);
+});
+
 socket.on("gameReady", function(playerData) {
-    console.log(playerData);
     player.id = playerData.id;
     player.pos = playerData.pos;
-    console.log("player position: ", [playerData.x, playerData.y]);
-    player.pos
+    console.log(playerData, player.pos);
 })
 
 socket.on("playersArray", function(playersArray){
     otherPlayers = playersArray;
     otherPlayers.forEach(function(player){
         player.sprite = new Sprite('img/capguy-walk.png', [0, 0], [184, 325], 16, [0, 1, 2, 3, 4, 5, 6, 7]);
-    })
+    });
+    console.log(otherPlayers);
+});
+
+socket.on('otherPlayerDC', function (socketId) {
+    var deletion = [];
+    otherPlayers.forEach(function (player, index) {
+        if (player.id === socketId) deletion.push(index);
+    });
+    deletion.forEach(function (index) {
+        otherPlayers.splice(index, 1);
+    });
```

Comment:
deletes disconnected players from field. iterates through the array to find players with the same socket.id as the disconnected player. Then splices them from the array.